app/views/ProfileView/index.js

```diff
@@ -181,7 +187,27 @@ class ProfileView extends React.Component {
 
 		// Username
 		if (user.username !== username) {
-			params.username = username;
+			const regEx = new RegExp(`^${ UTF8_Names_Validation }$`);
+			if (regEx.test(username)) {
```

Comment:
instead of make this logic here, I think it's better if you create a function like: `validUsername`.
```javascript
validUsername = () => {
  const { username } = this.state;
  const { UTF8_Names_Validation } = this.props;
  const usernameRegex = new RegExp(`^${ UTF8_Names_Validation }$`);
  if (usernameRegex.test(username)) {
    return true;
  }
  showErrorAlert(yourMessage, yourTitle);
  return false;
}
```
so, you can use only: 
```javascript
if (this.validUsername()) {
  params.username = username;
} else {
  this.setState({ saving: false });
  return;
}
```

Reply:
Yeah I agree on that, it'll be cleaner than